components/brave_wallet_ui/ledger/ledger.ts

```diff
@@ -0,0 +1,31 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import { LedgerCommand, LEDGER_BRIDGE_URL } from '../common/hardware/ledgerjs/ledger-messages'
+import { LedgerUntrustedMessagingTransport } from '../common/hardware/ledgerjs/ledger-untrusted-transport'
+
+const setUpAuthorizeButtonListner = (targetUrl: string) => {
+  const untrustedMessagingTransport = new LedgerUntrustedMessagingTransport(window.parent, targetUrl)
+  window.addEventListener('DOMContentLoaded', (event) => {
+    const authorizeBtn = document.getElementById('authorize')
+    if (authorizeBtn) {
+      authorizeBtn.addEventListener('click', () => {
+        untrustedMessagingTransport.promptAuthorization().then((result) => {
+          untrustedMessagingTransport.sendCommand({
+            id: LedgerCommand.AuthorizationSuccess,
+            origin: LEDGER_BRIDGE_URL,
+            command: LedgerCommand.AuthorizationSuccess
+          })
+        })
+        // TODO handle authorize failure / deny, button disable
```

Comment:
++

Reply:
For the case when the user clicks 'deny' on the authorization prompt, this error is logged to the console:
```
CustomError { name: 'TransportOpenUserCancelled', message: 'Access denied to use Ledger device' }
```
and the user is returned to the authorize screen.  Now that I look at this, that actually sounds like the right behavior to me.  Thoughts?

Reply:
sounds fine to me